packages/eas-build-job/src/metadata.ts

```diff
@@ -9,9 +9,9 @@ export enum FingerprintSourceType {
 }
 
 export type FingerprintSource =
-  | { type: FingerprintSourceType.GCS; bucketKey: string }
-  | { type: FingerprintSourceType.PATH; path: string }
-  | { type: FingerprintSourceType.URL; url: string };
+  | { type: FingerprintSourceType.GCS; bucketKey: string; isDebugFingerprint?: boolean }
+  | { type: FingerprintSourceType.PATH; path: string; isDebugFingerprint?: boolean }
+  | { type: FingerprintSourceType.URL; url: string; isDebugFingerprint?: boolean };
```

Comment:
```suggestion
export type FingerprintSource = { isDebugFingerprint?: boolean } &
  (
    | { type: FingerprintSourceType.GCS; bucketKey: string; }
    | { type: FingerprintSourceType.PATH; path: string; }
    | { type: FingerprintSourceType.URL; url: string; }
  );
```

nit: You can probably do something like this to avoid duplication